packages/perseus-core/src/utils/random_utils.ts

```diff
@@ -0,0 +1,77 @@
+/* Note(tamara): Brought over from the perseus package packages/perseus/src/util.ts file.
+    May be useful to bring other perseus package utilities here. Contains utility functions
+    and types used across multiple widgets for randomization and shuffling. */
+
+import _ from "underscore";
+
+type RNG = () => number;
+
+const seededRNG: (seed: number) => RNG = function (seed: number): RNG {
+    let randomSeed = seed;
+
+    return function () {
+        // Robert Jenkins' 32 bit integer hash function.
+        let seed = randomSeed;
+        seed = (seed + 0x7ed55d16 + (seed << 12)) & 0xffffffff;
+        seed = (seed ^ 0xc761c23c ^ (seed >>> 19)) & 0xffffffff;
+        seed = (seed + 0x165667b1 + (seed << 5)) & 0xffffffff;
+        seed = ((seed + 0xd3a2646c) ^ (seed << 9)) & 0xffffffff;
+        seed = (seed + 0xfd7046c5 + (seed << 3)) & 0xffffffff;
+        seed = (seed ^ 0xb55a4f09 ^ (seed >>> 16)) & 0xffffffff;
```

Comment:
This isn't your fault, but I strongly dislike having this kind of indecipherable bit manipulation code in our code base; borderline want to make a ticket to replace it with a small dep...but I guess it's fine for now.

Reply:
Yea, it seems a bit excessive, but it works so 🤷‍♀️